nemo_skills/inference/model/responses.py

```diff
@@ -0,0 +1,375 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from openai import AsyncOpenAI, OpenAI
+
+from nemo_skills.utils import get_logger_name
+
+from .base import BaseModel
+from .context_retry import with_context_retry
+
+LOG = logging.getLogger(get_logger_name(__file__))
+
+
+class ResponsesModel(BaseModel):
+    """Model implementation using OpenAI Responses API via LiteLLM.
+
+    This model uses the responses API endpoint instead of chat completions,
+    which is useful for models like gpt-oss that support the responses format.
+    """
+
+    MODEL_PROVIDER = "openai"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        # Initialize OpenAI clients for responses API
+        self.client = OpenAI(base_url=self.base_url, api_key=self.litellm_kwargs["api_key"])
+        self.async_client = AsyncOpenAI(base_url=self.base_url, api_key=self.litellm_kwargs["api_key"])
+
+    def _build_completion_request_params(
```

Comment:
Can we combine the logic here with `_build_chat_completion_request`, since it is largely repeated, except for the `stream` argument?